web/js/cartograph.js

```diff
@@ -238,6 +249,146 @@ CG.init = function(layer) {
         }
         var encoded = encodeURIComponent(title);
         var uri = 'https://en.wikipedia.org/w/api.php?action=query&format=json&titles=' + encoded + '&prop=pageimages|extracts&exintro&explaintext&exchars=400&callback=?';
+
+        var id = properties.id; //this will give the internal cartograph ids... FUN FACT: properties = id???
+        var relatedPoints = '../point.json?id=' + id;
+
+        //how I got the snippets: https://en.wikipedia.org/w/api.php?action=query&format=json&titles=Top%20Gun!!!!!!SOMETHING? extracts&explaintext&exsentences=3&exsectionformat=plain!!!!!!&callback=jQuery1102006687854900582613_1497380793059&_=1497380793065
+        //how I got the x,y coordinates of each link: look at RelatedPointsService.py
+
+        var randHue = 'rgb(' + (Math.floor(Math.random() * 256))
+                            + ',' + (Math.floor(Math.random() * 256))
+                            + ',' + (Math.floor(Math.random() * 256)) + ')';
+
+        var edgeHoverStyle =   {weight: 3,
+                                opacity: 0.95,
+                                smoothFactor: 1,
+                                attribution: 'edge'};
+        var edgeNeutralStyle = {color: randHue,
+                                weight: 1,
+                                opacity: 0.65,
+                                smoothFactor: 1,
+                                attribution: 'edge'};
+        $.getJSON(relatedPoints, function (data) {
+            CG.map.removeLayer(storeddotslayer);
+            CG.map.removeLayer(storedcurveslayer);
+
+            storeddots = [];
+            storedcurves = [];
+            storededges = [];
+
+            var smallMarker = L.icon({
+            iconUrl: 'images/blackDot.png',
+            iconSize: [10,10]
+            });
+
+            var coords = [];
+            var names = [];
+
+            data[id].forEach(function (linkInfo) {
+
+                  var x = linkInfo.data.loc[0];
+                  var y = linkInfo.data.loc[1];
+                  marker = L.marker([x, y], {icon: smallMarker});
+                  storeddots.push(marker);
+                  coords.push([linkInfo.data.id, linkInfo.data.name, [x, y]]);
+            });
+            storeddotslayer = L.layerGroup(storeddots);
+            CG.map.addLayer(storeddotslayer);
+
+
+            function createPathPairs(array){
```

Comment:
This function is just used below, so it would be better to remove it and put that logic directly below.